packages/core/src/http/HttpAgentImpl.ts

```diff
@@ -422,18 +422,15 @@ export class HttpAgentImpl extends HttpAgent {
       ...this._defaultRequestOptions,
       ...options,
     };
-    let cookieHeader = {};
 
-    if (composedOptions.withCredentials) {
-      cookieHeader = { Cookie: this._cookie.getCookiesStringForCookieHeader() };
+    if (composedOptions.fetchOptions?.credentials === 'include') {
+      // mock default browser behavior for server-side (sending auth cookie)
```

Comment:
I think this should also be deeply merged to retain the same functionality it had before:

```js
composedOptions.fetchOptions.headers = {
  ...{ Cookie: this._cookie.getCookiesStringForCookieHeader() },
  ...this._defaultRequestOptions?.fetchOptions?.headers,
  ...(options?.fetchOptions?.headers ?? {}),
};
```

Since now, if you have any cookies in default options or request options, they are overriden by the "credentials" cookie.

Reply:
Fixed, but... this should probably *combine* the `Cookie` string, instead of keeping one version, shouldn't it? E.g.:
`Cookie: 'ck1=foo; ck2=bar'` + `Cookie: 'ck3=baz'` = `Cookie: 'ck1=foo; ck2=bar; ck3=baz'` Otherwise the logic doesn't work as expected.

:grimacing: I'll look at it again.

Reply:
Done, cookies now combine.